### PR TITLE
Migration to Symfony 5.4

### PR DESCRIPTION
**Pull request by nickbass72**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,17 +16,17 @@
...
-        "symfony/finder": "^3.0 || ^4.0",
-        "symfony/console": "^3.0  || ^4.0",
-        "symfony/http-kernel": "^3.0 || ^4.0"
+        "doctrine/annotations": "^1.14",
+        "symfony/finder": "^5.0",
+        "symfony/console": "^5.0",
+        "symfony/http-kernel": "^5.0"
     },
     "require-dev": {
-        "symfony/symfony": "^3.0 || ^4.0",
```
Comment (mSprunskas) [inline composer.json:25]: Keep the support for older versions if possible

**Review by mSprunskas — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (nickbass72) [reply composer.json:25]: We had a long discussion about that. The problem is that there are breaking changes when requiring Symfony 5 because it requires Doctrine 2, etc. It is now being reviewed by one of my colleagues. The idea is to check if that complex support of multiple Symfony versions is possible. If not, we will create a new major version of this library.

Comment (daniel-valchev-paysera) [reply composer.json:25]: We discussed the issue and decided to drop the support for older Symfony versions as it will add too much complexity to the codebase.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,17 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## 3.0.0
+- Added
```
Comment (daniel-valchev-paysera) [inline CHANGELOG.md]: Let's use ## for the type of change (Added/Removed/Changed/Fixed/etc)

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,18 @@ All notable changes to this project will be documented in this file.
...
+  - Support for Symfony 5
+  - Support for PHP 8
+  - new command `bin/devenv` for creating dev environment for testing with php7.4, php8.0, php8.1. See README for details.
```
Comment (daniel-valchev-paysera) [inline CHANGELOG.md]: No need to put this in the changelog.

File: CHANGELOG.md [hunk 4]
```diff
@@ -25,7 +37,7 @@ paysera_database_init:
          fixtures:
              main: '%kernel.root_dir%/Fixtures'
          structure: *initial
-    exports:
```
Comment (daniel-valchev-paysera) [inline CHANGELOG.md:28]: Unnecessary change

File: .gitignore [hunk 5]
```diff
@@ -1,5 +1,9 @@
 /vendor/
-composer.lock
```
Comment (daniel-valchev-paysera) [inline .gitignore]: Unnecessary change

File: .gitignore [hunk 6]
```diff
@@ -1,5 +1,9 @@
...
+composer.lock*
 /tests/symfony/cache/
 /tests/symfony/logs/
+/docker/mount/*
+!/docker/mount/.gitkeep
```
Comment (daniel-valchev-paysera) [inline .gitignore]: Unnecessary change

File: src/Service/Exporter/StructureExporter.php [hunk 7]
```diff
@@ -3,21 +3,27 @@
...
 
-use \InvalidArgumentException;
+use InvalidArgumentException;
 use Paysera\Bundle\DatabaseInitBundle\Service\SqlDumperInterface;
 use Symfony\Component\Filesystem\Exception\IOException;
 
 class StructureExporter implements DatabaseExporterInterface
 {
+    /**
+     * @var SqlDumperInterface
```
Comment (daniel-valchev-paysera) [inline src/Service/Exporter/StructureExporter.php]: We can add the property type here instead of the phpdoc.

File: src/Service/Exporter/StructureExporter.php [hunk 8]
```diff
@@ -3,21 +3,27 @@
...
+     */
     private $dumper;
+    /**
+     * @var string
```
Comment (daniel-valchev-paysera) [inline src/Service/Exporter/StructureExporter.php]: We can add the property type here instead of the phpdoc.

**Review by daniel-valchev-paysera — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (nickbass72) [reply CHANGELOG.md:28]: There was an extra space character here, so I removed it.

**Review by daniel-valchev-paysera — APPROVED** [review]
